main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/csv"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type Pokemon struct {
+	Id      int    `json:"id"`
+	Name    string `json:"name"`
+	TypeOne string `json:"typeOne"`
+}
+
+func main() {
+
+	pokemons := initializePokemonData()
+
+	r := mux.NewRouter()
+	r.HandleFunc("/pokemon/{id}", func(rw http.ResponseWriter, r *http.Request) {
+		HandlePokemonById(rw, r, pokemons)
+	})
+	http.ListenAndServe(":8080", r)
+}
+
+func initializePokemonData() []Pokemon {
```

Comment:
I would also recommend moving all of the logic that takes care of reading the data on its own dedicated package.

Reply:
By doing this, then your handler would call this function